crates/subspace-gateway/src/commands/http/server.rs

```diff
@@ -0,0 +1,120 @@
+use actix_web::{web, App, HttpResponse, HttpServer, Responder};
+use serde::{Deserialize, Deserializer, Serialize};
+use std::default::Default;
+use std::error::Error;
+use std::sync::Arc;
+use subspace_data_retrieval::object_fetcher::ObjectFetcher;
+use subspace_data_retrieval::piece_getter::ObjectPieceGetter;
+use tracing::{debug, error, trace};
+
+pub(crate) struct ServerParameters<PG>
+where
+    PG: ObjectPieceGetter + Send + Sync + 'static,
+{
+    pub(crate) object_fetcher: ObjectFetcher<PG>,
+    pub(crate) indexer_endpoint: String,
+    pub(crate) http_endpoint: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+struct ObjectMapping {
+    #[serde(rename = "hash")]
+    hash: String,
+    #[serde(rename = "pieceIndex")]
+    piece_index: u64,
+    #[serde(rename = "pieceOffset")]
+    piece_offset: u32,
```

Comment:
This looks like `GlobalObject`, would it be easier to rename the fields, and re-use that type with `#[serde(flatten)]`?
https://github.com/autonomys/subspace/blob/aa85357117c0cefb7b9e33f7a5065ea271964843/crates/subspace-core-primitives/src/objects.rs#L99-L107

If not, let’s import the underlying types and use them.

Reply:
It seems like spreading the dependency across contexts. I would prefer changing types.

Reply:
Changing to `Blake3Hash` here might simplify some of the code below.

Reply:
Looks good, but I wanted to go a bit further, see [`02c0ca4` (#3293)](https://github.com/autonomys/subspace/pull/3293/commits/02c0ca4f1827d23034c9bfa0f43d806a6409042c) in #3293.